### PR TITLE
nixosTests.gnome3: port to python

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by worldofpeace** [comment]
@GrahamcOfBorg test gnome3

**Comment by worldofpeace** [comment]
Subtest is pretty unhelpful, I don't think I know what test failed...

**Comment by worldofpeace** [comment]
Ok, with new logging I can see what test exactly failed:
```
The following tests have failed:
 - Test "Login to GNOME with GDM" failed with error: "retrieving systemctl info for unit "default.target" under user "alice" failed with exit code 1"
cleaning up
```


maybe https://github.com/NixOS/nixpkgs/pull/76157/commits/e4b5bc179a953df2659e3f94960985d4c19b4368 does the trick on ofborg too.

**Comment by worldofpeace** [comment]
@GrahamcOfBorg test gnome3

**Comment by worldofpeace** [comment]
Uhhmm, aarch64 is broken for gnome. Will fix that but the test passes now.

**Comment by worldofpeace** [comment]
https://github.com/NixOS/nixpkgs/pull/78846

**Comment by worldofpeace** [comment]
@tfc @flokli @adisbladis

File: nixos/tests/gnome3.nix [hunk 1]
```diff
@@ -24,41 +24,53 @@ import ./make-test.nix ({ pkgs, ...} : {
...
-      # wait to get a nice screenshot
-      $machine->sleep(20);
-      $machine->screenshot("screen");
+      with subtest("Open Gnome Terminal"):
+          machine.succeed(
+              "${gnomeTerminalCommand}"
+          )
+          # correct output should be (true, '"Gnome-terminal"')
+          machine.wait_until_succeeds(
+              "${wmClass} | grep -q 'gnome-terminal-server'"
```
Comment (tfc) [inline nixos/tests/gnome3.nix]: somehow this output does not match the comment.

Comment (worldofpeace) [reply nixos/tests/gnome3.nix]: yep, that's a copy and paste error for the comment. Under wayland it's a different wmclass

Comment (tfc) [reply nixos/tests/gnome3.nix]: Can you align the comment to the reality? And then i give my thumbs up for review.

Comment (worldofpeace) [reply nixos/tests/gnome3.nix]: Done

**Review by jtojnar — APPROVED** [review]
Changes look good at a glance.

**Pull request merged**